latis-python-client/client/latis.py

```diff
@@ -0,0 +1,112 @@
+import numpy
+import pandas as pd
+import requests
+import urllib.parse
+
+
+class LatisInstance:
+
+    def __init__(self, baseUrl, latis3):
+        self.baseUrl = baseUrl
+        self.latis3 = latis3
+        self.__formatBaseUrl()
+
+        self.catalog = self.__createCatalog()
+
+    def createDataset(self, name):
+        return Dataset(self, name)
+
+    def __formatBaseUrl(self):
+        if not self.baseUrl[-1] == '/':
+            self.baseUrl += '/'
+        if self.latis3:
+            self.baseUrl += 'dap2/'
+        else:
+            self.baseUrl += 'dap/'
+
+    def __createCatalog(self):
+        return Catalog(self)
+
+
+class Catalog:
+
+    def __init__(self, latisInstance):
+
+        self.catalog = {}
+
+        if latisInstance.latis3:
+            js = requests.get(latisInstance.baseUrl).json()
+            dataset = js['dataset']
+            titles = numpy.array([k['title'] for k in dataset])
+            self.list = numpy.array([k['identifier'] for k in dataset])
+            for i in range(len(self.list)):
+                self.catalog[titles[i]] = self.list[i]
+        else:
+            q = latisInstance.baseUrl + 'catalog.csv'
+            df = pd.read_csv(q)
+            names = df['name']
+            self.list = df['accessURL'].to_numpy()
+            for i in range(len(self.list)):
+                self.catalog[names[i]] = self.list[i]
+
+    def search(self, searchTerm):
```

Comment:
What about `filter` returning a new Catalog?
Or `findDataset` that may or may not return a single matching Dataset. Does python have something like `option`? If not, I suppose `catalog.getDataset` with a potential error would suffice. Though `latis.getDataset` (delegating here) might be the way to go.